File: 383-ransom-note/ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int> mp;
        for(int i=0;i<magazine.size();i++){
            mp[magazine[i]]++;
        }
        for(int i=0;i<ransomNote.size();i++){
            char c = ransomNote[i];
            if (mp[c] > 0) {
                mp[c]--;
            }
            else
            return false;
        }
        return true;
        
    }
};

File: 383-ransom-note/ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        dict1={}


        for i in magazine:
            dict1[i]=dict1.get(i,0)+1

        for i in ransomNote:

            if i in dict1:
                dict1[i]-=1
                if dict1[i]==0:
                    del dict1[i]
            else:
                return False

        return True
        